src/phoebus_utils/history.hpp

```diff
@@ -0,0 +1,78 @@
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#ifndef PHOEBUS_UTILS_HISTORY_HPP_
+#define PHOEBUS_UTILS_HISTORY_HPP_
+
+#include <string>
+#include <vector>
+
+#include <kokkos_abstraction.hpp>
+#include <parthenon/package.hpp>
+#include <utils/error_checking.hpp>
+
+using namespace parthenon::package::prelude;
+
+/*
+ * History Template to build off of. Applies Reducer to MeshData on
+ * variable with varname. If variable is multi-d use idx to offset
+ * from first element. You should capture this object in a lambda to
+ * pass it to the Parthenon history machinery.
+ *
+ * TODO(JMM): This one is very simple. No volume weighting, no metric
+ * information. So it only works for min and max basically.  This can
```

Comment:
doenst it work for volume integrals of densitized conserved quantities?

Reply:
oops yes it does. This comment is old.